### PR TITLE
SNOW-1570506 Fix `Series.argmax` and `Series.argmin`, add tests

### PR DESCRIPTION
**Pull request by sfc-gh-vbudati**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1570506

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

 Fixed the existing implementation of `Series.argmax` and `Series.argmin`. Previously, `np.isnan(result)` was not working since the wrong type of `result` was being provided. Instead, I changed this logic to directly see if result is some form of None/nan, and if it is return -1 (valid result does not exist).

```py
>>> s = pd.Series([13, 1, 2, None, 1], index=["A", "B", "C", "D", "E"])
>>> s.argmin(skipna=False)
-1
>>> s.argmin(skipna=True)
1
>>> s.argmax(skipna=False)
-1
>>> s.argmax(skipna=True)
0
```

### CODE REVIEW

**Review by sfc-gh-nkrishna — APPROVED** [review]
LGTM!

File: src/snowflake/snowpark/modin/pandas/series.py [hunk 1]
```diff
@@ -781,25 +781,37 @@ def apply(
...
-        if np.isnan(result) or result is pandas.NA:
+        if self._query_compiler.has_multiindex():
+            # The index is a MultiIndex, current logic does not support this.
+            ErrorMessage.not_implemented(
+                "Series.argmax is not yet supported when the index is a MultiIndex."
+            )
+        result = self.reset_index(drop=True).idxmax(
+            axis=axis, skipna=skipna, *args, **kwargs
+        )
+        if not is_integer(result) and not result:  # if result=0, return 0 not -1
```
Comment (sfc-gh-evandenberg) [inline src/snowflake/snowpark/modin/pandas/series.py:797]: Is this comment accurate?  It says if result=0 but I don't see that as a check here.

**Review by sfc-gh-evandenberg — APPROVED** [review]
LGTM with one question.

File: src/snowflake/snowpark/modin/pandas/series.py (hunk 1 above)
Comment (sfc-gh-helmeleegy) [reply src/snowflake/snowpark/modin/pandas/series.py:797]: The result=0 case is not captured by this condition, and so it's simply handled by the `return result` line below. What I'm not clear about is why one would think that if result=0, we return -1? Maybe remove this comment?

Comment (sfc-gh-helmeleegy) [inline src/snowflake/snowpark/modin/pandas/series.py:797]: Why `and` and not `or` like the original implementation?

File: src/snowflake/snowpark/modin/pandas/series.py [hunk 2]
```diff
@@ -781,25 +781,37 @@ def apply(
...
         """
         # TODO: SNOW-1063347: Modin upgrade - modin.pandas.Series functions
-        result = self.idxmax(axis=axis, skipna=skipna, *args, **kwargs)
-        if np.isnan(result) or result is pandas.NA:
+        if self._query_compiler.has_multiindex():
+            # The index is a MultiIndex, current logic does not support this.
+            ErrorMessage.not_implemented(
+                "Series.argmax is not yet supported when the index is a MultiIndex."
+            )
+        result = self.reset_index(drop=True).idxmax(
```
Comment (sfc-gh-helmeleegy) [inline src/snowflake/snowpark/modin/pandas/series.py:794]: Given that we are calling `reset_index` here, it seems that `result` can never be None or non-int. So I'm wondering if the check below is ever satisfied.

Comment (sfc-gh-vbudati) [reply src/snowflake/snowpark/modin/pandas/series.py:794]: result is None or a non-integer when idxmax does not find a matching value, so the check still needs to exist

File: src/snowflake/snowpark/modin/pandas/series.py (hunk 1 above)
Comment (sfc-gh-vbudati) [reply src/snowflake/snowpark/modin/pandas/series.py:797]: I think I should be able to reduce to just check `not is_integer(result)`

Comment (sfc-gh-vbudati) [reply src/snowflake/snowpark/modin/pandas/series.py:797]: I'm not directly checking whether result is 0, rather checking if result is a valid integer. I should be able to reduce this check to just `not is_integer(result)` and it should work fine

**Review by sfc-gh-helmeleegy — APPROVED** [review]
Looks great, thanks Varnika!

**Pull request merged**